#!/usr/bin/env python3
PACKAGE = "mediassist3_densemap"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("prefilter", 				bool_t, 0, "Enable/Disable all Prefilters", True)
gen.add("prefilter_segmentation", 	bool_t, 0, "Enable/Disable Segmantation prefilter", True)
gen.add("prefilter_color", 			bool_t, 0, "Enable/Disable Color prefilter", True)
gen.add("prefilter_distance", 		bool_t, 0, "Enable/Disable CamDistance prefilter", True)

gen.add("segmentation_count", int_t, 0, "segmetaion parameter", 2, 1, 10)
gen.add("hsv_value", double_t, 0, "value for hsv filter", 0.75, 0, 1)
gen.add("dist_min", double_t,  0, "min distance for cam distance filter", 0.1, 0, 1)
gen.add("dist_max", double_t,  0, "max distance for cam distance filter", 0.2, 0, 1)

gen.add("statistical_outlier", 			 bool_t, 0, "Enable/Disable statistical outlier filter",True)
gen.add("voxel_grid_filter", 			 bool_t, 0, "Enable/Disable voxel_grid_filter", False)
gen.add("approximate_voxel_grid_filter", bool_t, 0, "Enable/Disable approximate voxel grid filter", True)
gen.add("fullshuffle", bool_t, 0, "switch fullshuffle and small shuffle", False)


exit(gen.generate(PACKAGE, "mediassist3_densemap", "dense_map_node"))